hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -785,7 +760,7 @@ public boolean merge(Data key, EntryView mergingEntry, MapMergePolicy mergePolic
             }
             newValue = mapDataStore.add(key, newValue, now);
             onStore(record);
-            recordFactory.setValue(record, newValue);
+            storage.updateRecordValue(key, record, newValue);
```

Comment:
@ahmetmircik can you please check here ? I've moved updating the value to storage